### PR TITLE
feat: adding complaint/feedback form.

### PR DESCRIPTION
**Pull request by jonasbrunvoll**
This PR includes the following:

- Feedback layout.
- Content page for /contact/billettkontroll-og-gebyr/tilbakemelding
- xState state machine for /contact/billettkontroll-og-gebyr/tilbakemelding.
- More translations rin contact.ts.
- minor updates of the input component, like validation of input.

### CODE REVIEW

File: src/page-modules/contact/components/input/input.tsx [hunk 1]
```diff
@@ -5,24 +5,46 @@ import { andIf } from '@atb/utils/css';
...
       })}
     >
-      <label className={style.label}>{t(label)}</label>
+      <div>
+        <label>{t(label)}</label>
+        {validationMessage && (
+          <>
+            {' - '}
+            <label className={style.error}>{t(validationMessage)}</label>
+          </>
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/input.tsx:36]:
```suggestion
          <label className={style.error}>{t(validationMessage)}</label>
```

File: src/page-modules/contact/components/input/input.tsx [hunk 2]
```diff
@@ -5,24 +5,46 @@ import { andIf } from '@atb/utils/css';
...
 type InputProps = {
   label: TranslatedString;
+  validationMessage?: TranslatedString;
   onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
 } & JSX.IntrinsicElements['input'];
 
-export const Input = ({ label, type, name, checked, onChange }: InputProps) => {
+export const Input = ({
+  label,
+  validationMessage,
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/input.tsx:14]:
```suggestion
  errorMessage,
```

File: src/page-modules/contact/components/input/input.module.css [hunk 3]
```diff
@@ -36,3 +35,11 @@
...
     var(--interactive-interactive_2-outline-background);
 }
+
+.error {
+  color: red;
+}
+
+.input_error {
+  border-color: red;
+}
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/input.module.css:45]: To align more with other css files I suggest to use: 
```suggestion
.input_label__error {
  color: red;
}

.input__error {
  border-color: red;
}
```

File: src/page-modules/contact/layouts/ticket-control-page-layout.tsx [hunk 4]
```diff
@@ -35,7 +35,7 @@ function ComplaintPageLayout({ children }: ComplaintPageLayoutProps) {
             checked={router.pathname.includes('/utsette-betaling')}
             onChange={() =>
               router.push(
-                '/contact/billettkontroll-og-gebyr/klage',
+                '/contact/billettkontroll-og-gebyr/utsette-betaling',
```
Comment (mortennordseth) [inline src/page-modules/contact/layouts/ticket-control-page-layout.tsx:38]: You don't have to do anything about it in this PR, but we should probably use `kontakt` instead of `contact` when the rest of the URL is in Norwegian

File: src/page-modules/contact/lines/use-lines.ts [hunk 5]
```diff
@@ -17,7 +17,8 @@ export const useLines = () => {
 
   const getQuaysByLine = (lineId: string): Line['quays'] => {
     if (!lines) return [];
-    return lines.find((line) => line.id === lineId)?.quays || [];
```
Comment (mortennordseth) [inline src/page-modules/contact/lines/use-lines.ts:20]: Unnecessary change, but it probably stems from logging activity

File: src/translations/pages/contact.ts [hunk 6]
```diff
@@ -158,6 +158,74 @@ export const Contact = {
...
+        label: _('Reisemåte', 'Transport mode', 'Reisemåte'),
+        optionLabel: _(
+          'Velg reisemåte',
+          'Choose transport mode',
+          'Velg reisemåte',
+        ),
+        notDefinedFeedback: _(
+          'Reisemåte må settes!',
+          'Transport mode must be set!',
+          'Reisemåte må settes!',
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:188]: I don't think we need "!" here. Let's try with something like: `Velg et transportmiddel` / `Select a transport mode`. This applies to the other error messages as well

File: src/translations/pages/contact.ts [hunk 7]
```diff
@@ -158,6 +158,74 @@ export const Contact = {
...
+        'Kor var billettkontrollen?',
+      ),
+      transportMode: {
+        label: _('Reisemåte', 'Transport mode', 'Reisemåte'),
+        optionLabel: _(
+          'Velg reisemåte',
+          'Choose transport mode',
+          'Velg reisemåte',
+        ),
+        notDefinedFeedback: _(
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:185]:
```suggestion
        errorMessage: _(
```

File: src/translations/pages/contact.ts [hunk 8]
```diff
@@ -158,6 +158,74 @@ export const Contact = {
...
+        ),
+      },
+      departureLocation: {
+        label: _('Avreisested', 'Departure location', 'Avreisested'),
+        optionLabel: _(
+          'Velg avreisested',
+          'Choose departure loation',
+          'Velg avreisested',
+        ),
+        notDefinedFeedback: _(
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:207]:
```suggestion
        errorMessage: _(
```

File: src/translations/pages/contact.ts [hunk 9]
```diff
@@ -158,6 +158,74 @@ export const Contact = {
...
+        ),
+      },
+      arrivalLocation: {
+        label: _('Ankomststed', 'Arrival location', 'Ankomststed'),
+        optionLabel: _(
+          'Velg anskomststed',
+          'Choose arrival location',
+          'Velg ankomststed',
+        ),
+        notDefinedFeedback: _(
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:220]:
```suggestion
        errorMessage: _(
```

File: src/translations/pages/contact.ts [hunk 10]
```diff
@@ -178,29 +246,39 @@ export const Contact = {
...
-    'What do you want to tell us?',
-    'Kva ønskjer du å fortelje oss?',
-  ),
+  feedback: {
+    question: _(
+      'Hva ønsker du å fortelle oss?',
+      'What do you want to tell us?',
+      'Kva ønskjer du å fortelje oss?',
+    ),
+    isToShort: _(
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:255]:
```suggestion
    errorMessage: _(
```

File: src/translations/pages/contact.ts [hunk 11]
```diff
@@ -178,29 +246,39 @@ export const Contact = {
...
     ),
-    surname: _('Etternavn', 'Surname', 'Etternamn'),
+    lastname: _('Etternavn', 'Surname', 'Etternamn'),
     address: _('Adresse', 'Address', 'Adresse'),
     postalCode: _('Postnummer', 'Postal code', 'Postnummar'),
     city: _('Bosted', 'City', 'Bustad'),
     email: _('E-post', 'Email', 'E-post'),
     mobile: _('Mobil', 'Mobile', 'Mobil'),
     bankAccount: _('Bankkontonummer', 'Bank account number', 'Bankkontonummar'),
+    isEmpty: _('Må settes!', 'Must be set!', 'Må settes!'),
```
Comment (mortennordseth) [inline src/translations/pages/contact.ts:281]:
```suggestion
    errorMessage: _('Vennligst fyll ut informasjon om deg', 'Please provide your information', 'Vennligst fyll ut din informasjon'),
```

File: src/page-modules/contact/ticket-control/ticket-control.module.css [hunk 12]
```diff
@@ -24,3 +24,11 @@ ul {
   box-shadow: inset 0 0 0 var(--border-width-medium)
     var(--interactive-interactive_2-outline-background);
 }
+
+.errorTextarea {
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/ticket-control.module.css:28]:
```suggestion
.feedback__error {
```

File: src/page-modules/contact/ticket-control/ticket-control.module.css [hunk 13]
```diff
@@ -24,3 +24,11 @@ ul {
...
+  border-color: red;
+}
+
+.error {
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/ticket-control.module.css:32]:
```suggestion
.feedback_label__error {
```

File: src/page-modules/contact/ticket-control/feedback/feedbackForm.tsx [hunk 14]
```diff
@@ -0,0 +1,390 @@
...
+
+    const { user, subject, description } = fromContextFeedbackFormToTicketData(
+      state.context as ContextFeedbackForm,
+    );
+
+    const ticketData: TicketData = {
+      subject: subject,
+      description: description,
+      user: user,
+    };
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/feedback/feedbackForm.tsx:46]: I think that the `fromContextFeedbackFormToTicketData` should return an object of the type `TicketData` since its name `ToTicketData`

File: src/page-modules/contact/ticket-control/feedback/feedbackForm.tsx [hunk 15]
```diff
@@ -0,0 +1,390 @@
...
+
+    console.log({ ticketData });
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/feedback/feedbackForm.tsx:48]:
```suggestion
```

File: src/page-modules/contact/ticket-control/feedback/feedbackForm.tsx [hunk 16]
```diff
@@ -0,0 +1,390 @@
...
+
+    const response = await fetch('/api/contact', {
+      method: 'POST',
+      body: null,
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/feedback/feedbackForm.tsx:52]:
```suggestion
      body: JSON.stringify(ticketData),
```

File: src/page-modules/contact/ticket-control/feedback/feedbackForm.tsx [hunk 17]
```diff
@@ -0,0 +1,390 @@
...
+          name="transportModes"
+          value={state.context.transportMode}
+          onChange={(e) =>
+            send({
+              type: 'SET_TRANSPORT_MODE',
+              transportMode: e.target.value as TransportModeType,
+            })
+          }
+        >
+          <option
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/feedback/feedbackForm.tsx:110]: Should this option have the `disabled` tag?

**Review by mortennordseth — COMMENTED** [review]
A lot of good work here. I've added some suggestions and general feedback

File: src/page-modules/contact/layouts/ticket-control-page-layout.tsx (hunk 4 above)
Comment (jonasbrunvoll) [reply src/page-modules/contact/layouts/ticket-control-page-layout.tsx:38]: Agreed👍

File: src/page-modules/contact/components/input/input.tsx [hunk 18]
```diff
@@ -5,26 +5,52 @@ import { andIf } from '@atb/utils/css';
...
         [style.container]: true,
-        [style.rowDisplay]: type === 'radio' || type === 'checkbox',
+        [style.rowDisplay]:
+          type === 'radio' || type === 'checkbox' || type === 'submit',
       })}
     >
-      <label className={style.label}>{t(label)}</label>
+      <div>
+        <label>{t(label)}</label>
+      </div>
```
Comment (mortennordseth) [inline src/page-modules/contact/components/input/input.tsx:32]: Div not needed here afaik

File: src/page-modules/contact/ticket-control/feedback/feedbackForm.tsx [hunk 19]
```diff
@@ -0,0 +1,388 @@
...
+            label={t(
+              PageText.Contact.ticketControl.feedback.transportMode.optionLabel,
+            )}
+            disabled
```
Comment (mortennordseth) [inline src/page-modules/contact/ticket-control/feedback/feedbackForm.tsx:114]: From our testing I believe we found it best to have `selected` as well?

**Review by mortennordseth — APPROVED** [review]
